sql/sql_acl.cc

```diff
@@ -14312,6 +14313,22 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else if (redirect_enabled)
+  {
+    char msg[MAX_REDIRECTION_LEN];
+    size_t len = 0;
+
+    msg[0] = '\0';
+    len = snprintf(msg + 1, MAX_REDIRECTION_LEN - 1, "Location: mysql://[%s]:%s/user=%s&ttl=%s", 
```

Comment:
```suggestion
    len = snprintf(msg + 1, MAX_REDIRECTION_LEN - 1, "Location: mysql://[%s]:%s/user=%s&ttl=%s\n", 
```

As per [my Jira comment](https://jira.mariadb.org/browse/MDEV-15935?focusedCommentId=148949&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148949), I'd strongly urge us to add a terminator to this string, e.g., `\n`.

That will cause clients to process the string only up until the terminator character, not to the end of the OK packet. This will give us the flexibility to add additional information to this "human readable status information" string in the future without breaking existing clients.